hack/gen_versions_map.sh

```diff
@@ -30,7 +30,7 @@ resolved_miss_map=$(
     fi
 
     # if commit is not HEAD, check if it's valid
-    if [ $commit != "HEAD" ]; then
+    if [ "$commit" != "HEAD" ]; then
```

Comment:
This is a common pattern:
```suggestion
    if [ "x$commit" != "xHEAD" ]; then
```